### ISSUE TITLE
player.connect() doesn't connect to proxy mode

### ISSUE THREAD

**Issue by pingpong1109** [benbaptist/minecraft-wrapper#169]
I have online mod disabled and it still says this when I connect to the proxy mode with that function
![halp](https://cloud.githubusercontent.com/assets/751127/6883693/b6bbdfc6-d58f-11e4-9600-5e20463476fe.PNG)

**Comment by pingpong1109** [comment]
#Minecraft server properties
#Sat Mar 28 20:48:36 EDT 2015
spawn-protection=16
max-tick-time=60000
generator-settings=
force-gamemode=false
allow-nether=true
gamemode=1
enable-query=false
player-idle-timeout=0
difficulty=1
spawn-monsters=true
op-permission-level=4
resource-pack-hash=
announce-player-achievements=true
pvp=false
snooper-enabled=true
level-type=DEFAULT
hardcore=false
enable-command-block=true
max-players=20
network-compression-threshold=-1
max-world-size=29999984
server-port=25564
server-ip=127.0.0.1
spawn-npcs=true
allow-flight=false
level-name=world
view-distance=10
resource-pack=
spawn-animals=true
white-list=false
generate-structures=true
online-mode=false
max-build-height=256
level-seed=
motd=Platform Power
enable-rcon=false

Only differences are the motd and ports

**Comment by suresttexas00** [comment]
need the wrapper.properties too...

**Comment by suresttexas00** [comment]
and the wrapper/server properties of any other servers running on the same machine...

**Comment by pingpong1109** [comment]
[General]
server-name = Pingpong test Wrapper.py
command = C:\Program Files\Java\jre7\bin\java.exe -Xmx1024M -Xms1024M -jar minecraft_server.jar nogui
auto-restart = True
auto-update-wrapper = False
auto-update-dev-builds = False
pre-1.7-mode = False
timed-reboot = False
timed-reboot-seconds = 86400
timed-reboot-warning-minutes = 5
debug = True
shell-scripts = False
auto-update-dev-build = False

[Backups]
enabled = False
backup-folders = ['server.properties', 'world']
backup-interval = 3600
backup-notification = True
backup-location = backup-directory
backup-compression = False
backups-keep = 10

[IRC]
irc-enabled = True
server = 192.168.1.142
port = 6667
nick = Game
channels = ['#chat']
command-character = .
show-channel-server = True
autorun-irc-commands = ['']
obstruct-nicknames = False
control-from-irc = False
control-irc-pass = paul1234
password
show-irc-join-part = True

[Proxy]
proxy-enabled = True
proxy-port = 25563
proxy-bind = 0.0.0.0
server-port = 25564
online-mode = True
max-players = 1024
convert-player-files = False
spigot-mode = False

[Web]
web-enabled = True
web-bind = 127.0.0.1
web-port = 8071
web-password = paul1234
web-allow-file-management = True
public-stats = True

**Comment by pingpong1109** [comment]
I know super secure password :P

**Comment by pingpong1109** [comment]
second server files

**Comment by pingpong1109** [comment]
[General]
server-name = Pingpong test Wrapper.py
command = C:\Program Files\Java\jre7\bin\java.exe -Xmx1024M -Xms1024M -jar minecraft_server.jar nogui
auto-restart = True
auto-update-wrapper = False
auto-update-dev-builds = False
pre-1.7-mode = False
timed-reboot = False
timed-reboot-seconds = 86400
timed-reboot-warning-minutes = 5
debug = True
shell-scripts = False
auto-update-dev-build = False

[Backups]
enabled = False
backup-folders = ['server.properties', 'world']
backup-interval = 3600
backup-notification = True
backup-location = backup-directory
backup-compression = False
backups-keep = 10

[IRC]
irc-enabled = True
server = 192.168.1.142
port = 6667
nick = Lobby
channels = ['#chat']
command-character = .
show-channel-server = True
autorun-irc-commands = ['']
obstruct-nicknames = False
control-from-irc = False
control-irc-pass = paul1234
password
show-irc-join-part = True

[Proxy]
proxy-enabled = True
proxy-port = 25565
proxy-bind = 0.0.0.0
server-port = 25566
online-mode = True
max-players = 1024
convert-player-files = False
spigot-mode = False

[Web]
web-enabled = True
web-bind = 127.0.0.1
web-port = 8070
web-password = paul1234
web-allow-file-management = True
public-stats = True

**Comment by pingpong1109** [comment]
#Minecraft server properties
#Sat Mar 28 20:48:09 EDT 2015
spawn-protection=16
max-tick-time=60000
generator-settings=
force-gamemode=false
allow-nether=true
gamemode=1
enable-query=false
player-idle-timeout=0
difficulty=1
spawn-monsters=true
op-permission-level=4
resource-pack-hash=
announce-player-achievements=true
pvp=false
snooper-enabled=true
level-type=DEFAULT
hardcore=false
enable-command-block=true
max-players=20
network-compression-threshold=-1
max-world-size=29999984
server-port=25566
server-ip=127.0.0.1
spawn-npcs=true
allow-flight=false
level-name=world
view-distance=10
resource-pack=
spawn-animals=true
white-list=false
generate-structures=true
online-mode=false
max-build-height=256
level-seed=
enable-rcon=false
motd=Lobby

**Comment by suresttexas00** [comment]
Both of your IRCs are enabled on port 6667..?

Are you, by chance, using one as a lobby that transfers you to the other??

**Comment by suresttexas00** [comment]
Oh.. now i read the title.. LOL!  The second wrapper instance, Platform Power, has to run both the server.properties AND the wrapper.properties in offline mode... Lobby's wrapper will do all the authentication.  make sure Platform Power's port is not accessible from outside your private network, or people can connect to it directly, bypassing the Lobby.

**Comment by pingpong1109** [comment]
Yes one is a lobby and both use irc to combine chat. I can check the properties for one sayings online mode

**Comment by suresttexas00** [comment]
Don't know much about the IRC aspect, but if it's like most ported applications, I would not think the wrappers could both bind the same port (6667).  But, yes ALL servers and ALL wrappers must run OFFLINE except for the lobby wrapper.

**Comment by benbaptist** [comment]
The wrapper doesn't create its own IRC server - it just connects to one, so it can connect multiple times.

**Comment by suresttexas00** [comment]
Ben, I have a question... is the server.properties IP supposed to be set to 127.0.0.1 or left blank?

**Comment by benbaptist** [comment]
You can do whatever you want. It's probably best to do 127 for security purposes, or at least bind to a port inaccessible from the outside.

**Comment by suresttexas00** [comment]
Ben, does this actually work?  I tried doing this myself and it is not working.  Here is my server setups:
SERVER1 (the "lobby", although it is my main world):
Outside world URL -> port forward minecraft to 25566 (for my vanilla, anyway)
Wrapper proxy listening on port 25566  (online mode)
directs to server Jar listening to 25560  (OFFLINE mode)

Second server:
** no port forwards**
Wrapper proxy listening on port 25600  (OFFLINE mode)
directs to server Jar listening to 25610  (OFFLINE mode)
(I will include snippets of the properties files in a bit to verify I did not mistype something...)

Firstly, I was a bit confused on how to pass the arguments for player.connect() for a local, non-port forwarded server.  I also had questions about the data types the arguments are expecting... string, I guess?  I tried several different arguments and such.  Some I tried were:

most of these resulted in red bold letters "__Could not connect to that server!__"
`player.connect("127.0.0.1:", "25600")`
`player.connect("127.0.0.1", "25600")`
player.connect("play.surestcraft.com", "25600") (not that I expected that one to work)

This one gets me to at least attempt a connection:
![image](https://cloud.githubusercontent.com/assets/10001762/8611974/30d4bcc6-2697-11e5-97ab-e50b1073c372.png)
The destination server did not even see a connection attempt.

**Comment by suresttexas00** [comment]
Does this actually work?

I tried this (with another server running offline mode on port 25565):
player.connect(None, 25565)

```
[20:53:37] [Wrapper.py/INFO] Plugins reloaded
[20:53:38] [Wrapper.py/INFO] SurestTexas00 executed: /Well? (test server)
[20:53:38] [Wrapper.py/INFO] Disconnected from server: You logged in from another location
Sending self.clientSettings...
{'locale': 'en_US', 'chat_colors': False, 'displayed_skin_parts': 127, 'chat_mode': 0, 'view_distance':
16}
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 984, in handle
    id, original = self.packet.grabPacket()
  File "Wrapper\proxy.py", line 1033, in grabPacket
    length = self.unpack_varInt()
  File "Wrapper\proxy.py", line 1063, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "Wrapper\proxy.py", line 1215, in recv
    raise EOFError("Packet was zero length, disconnecting")
EOFError: Packet was zero length, disconnecting

Last packet IDs (Server->Client) before disconnection:
[('0x2', 52)]
Failed to grab packet (CLIENT):
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 588, in handle
    id, original = self.packet.grabPacket()
  File "Wrapper\proxy.py", line 1039, in grabPacket
    dataLength = self.unpack_varInt()
  File "Wrapper\proxy.py", line 1063, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "Wrapper\proxy.py", line 1213, in recv
    d = self.socket.recv(length)
  File "C:\Python27\lib\socket.py", line 170, in _dummy
    raise error(EBADF, 'Bad file descriptor')
error: [Errno 9] Bad file descriptor

[20:53:38] [Server thread/INFO]: SurestTexas00 lost connection: TextComponent{text='You logged in from a
nother location', siblings=[], style=Style{hasParent=false, color=null, bold=null, italic=null, underlin
ed=null, obfuscated=null, clickEvent=null, hoverEvent=null, insertion=null}}
[20:53:38] [Server thread/INFO]: ºbSurestTexas00ºr left the game
[20:53:38] [Server thread/INFO]: com.mojang.authlib.GameProfile@1b8c8319[id=3269fd15-5be9-3c2a-af6c-1214
7a760f78,name=SurestTexas00,properties={},legacy=false] (/127.0.0.1:52687) lost connection: Internal Exc
eption: io.netty.handler.codec.DecoderException: java.io.IOException: Bad packet id 21
Traceback (most recent call last):
  File "Wrapper\proxy.py", line 984, in handle
    id, original = self.packet.grabPacket()
  File "Wrapper\proxy.py", line 1033, in grabPacket
    length = self.unpack_varInt()
  File "Wrapper\proxy.py", line 1063, in unpack_varInt
    val = struct.unpack('B', self.recv(1))[0]
  File "Wrapper\proxy.py", line 1215, in recv
    raise EOFError("Packet was zero length, disconnecting")
EOFError: Packet was zero length, disconnecting

Last packet IDs (Server->Client) before disconnection:
[('0x19', 4), ('0x2f', 6), ('0x2f', 6), ('0x2f', 6), ('0x2f', 6), ('0x12', 9), ('0x15', 7), ('0x19', 4),
 ('0x19', 4), ('0x40', 39)]
```

**Comment by suresttexas00** [comment]
Getting close to fixing this.  We are going to need a wiki item to explain how to properly set up hubs.  Perhaps even have a sample server folder already configured.

**Comment by suresttexas00** [comment]
At this point, I have it working to this point:
1) you can log onto a server hub, type '/world1' and go to world1... 
2) however, you will be stuck and jerking around in the same place until you are killed.. then you respawn properly.
3) you can play in the world and then...
4) type '/lobby" and you get disconnected... but you CAN log back into the '/world1' again.
5) If you don't use /lobby to disconnect, strange things happen.. you log onto the world1, but your client sees the hub.. and other strange things.

basically need three things still fixed:
1) get player to respawn properly without dying first!
2) get the player to reconnect to the original server properly
3) get each serverconnnection to forget some of the prior settings.. the two servers/wrappers still get confused.

**Comment by suresttexas00** [comment]
#542
